src/adapter/src/catalog/builtin.rs

```diff
@@ -2271,6 +2288,186 @@ FROM mz_catalog.mz_roles r
 JOIN mz_catalog.mz_databases d ON (d.id IS NULL OR d.name = pg_catalog.current_database())",
 };
 
+pub const MZ_SHOW_MATERIALIZED_VIEWS: BuiltinView = BuiltinView {
+    name: "mz_show_materialized_views",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE VIEW mz_internal.mz_show_materialized_views
+AS SELECT mviews.name, clusters.name AS cluster, schema_id, cluster_id
+FROM mz_materialized_views AS mviews
+JOIN mz_clusters AS clusters ON clusters.id = mviews.cluster_id",
+};
+
+pub const MZ_SHOW_INDEXES: BuiltinView = BuiltinView {
+    name: "mz_show_indexes",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE VIEW mz_internal.mz_show_indexes
+AS SELECT
+    idxs.name AS name,
+    objs.name AS on,
+    clusters.name AS cluster,
+    COALESCE(keys.key, '{}'::_text) AS key,
+    idxs.on_id AS on_id,
+    objs.schema_id AS schema_id,
+    clusters.id AS cluster_id
+FROM
+    mz_indexes AS idxs
+    JOIN mz_catalog.mz_objects AS objs ON idxs.on_id = objs.id
+    JOIN mz_catalog.mz_clusters AS clusters ON clusters.id = idxs.cluster_id
+    LEFT JOIN
+        (SELECT
+            idxs.id,
+            ARRAY_AGG(
```

Comment:
Our docs specifically say that materializing `ARRAY_AGG` is an anti-pattern, https://materialize.com/docs/sql/functions/array_agg/#usage-in-dataflows. However, I don't think the suggestion helps us that much here. We could only create an index over something like
```
SELECT idxs.id, idx_cols.on_expression, obj_cols.name
FROM 
    mz_indexes AS idxs
    JOIN mz_index_columns idx_cols ON idxs.id = idx_cols.index_id
    LEFT JOIN mz_columns obj_cols ON
        idxs.on_id = obj_cols.id AND idx_cols.on_position = obj_cols.position
```
However then we'd still have to do an aggregation and a 4-way join every times we execute `SHOW INDEXES`.

Reply:
I think this is fine, despite the notes. This would be expensive to maintain if we had indexes whose keys have large numbers of expressions, and we continually add and remove individual expressions from each key. I think we have zero plan to do that, and instead only add entire collections at a time. This should be good, imo!